git-cliff-release/action.yaml

```diff
@@ -102,22 +114,42 @@ runs:
         GIT_CLIFF_REPOSITORY: ${{ github.workspace }}/__release_metadata_repo
         GIT_CLIFF_WORKDIR: ${{ github.action_path }}
         GITHUB_REPO: ${{ github.repository }}
-        GH_TOKEN: ${{ github.token }}
+        GITHUB_TOKEN: ${{ inputs.token }}
       working-directory: ${{ github.action_path }}
       run: |
+        set -x
+        dump_context_args=()
+        enhance_context_args=()
+        print_changelog_args=()
+
+        out_file=$GIT_CLIFF_WORKDIR/new_changelog.md
+
+        if [[ -n "${{ inputs.existing_changelog_path }}" ]]; then
+          out_file=$GIT_CLIFF_WORKDIR/existing_changelog.md
+          cp "${{ github.workspace }}/${{ inputs.existing_changelog_path }}" $out_file
+          sed '/<!-- git-cliff-unreleased-start -->/,/<!-- git-cliff-unreleased-end -->/d' -i $out_file
+
+          dump_context_args+=(--unreleased)
+          print_changelog_args+=(--prepend "$out_file" --unreleased)
+        else
+          print_changelog_args+=(--output "$out_file")
+        fi
+
         if [[ ${{ inputs.release_type }} = prerelease ]]; then
-          echo 'changelog<<EOF' >> $GITHUB_OUTPUT
-          git-cliff --context | 
-            python enhance_context.py --repo $GITHUB_REPO --unreleased-version "${{ steps.version_number.outputs.tag_name }}" | 
-            git-cliff --from-context - >> $GITHUB_OUTPUT
-          echo 'EOF' >> $GITHUB_OUTPUT
+          enhance_context_args+=(--unreleased-version "${{ steps.version_number.outputs.tag_name }}")
         else
-          echo 'changelog<<EOF' >> $GITHUB_OUTPUT
-          git-cliff --tag "${{ steps.version_number.outputs.tag_name }}" --context | 
-            python enhance_context.py --repo $GITHUB_REPO | 
-            git-cliff --from-context - >> $GITHUB_OUTPUT
-          echo 'EOF' >> $GITHUB_OUTPUT
+          dump_context_args+=(--tag "${{ steps.version_number.outputs.tag_name }}")
         fi
+
+        git-cliff --context "${dump_context_args[@]}" |
+          python enhance_context.py --repo $GITHUB_REPO "${enhance_context_args[@]}" |
+          git-cliff --from-context - "${print_changelog_args[@]}"
+
+        sed -i '$a\' $out_file # Make sure there is a newline at the end of the output file
```

Comment:
Does this work? Considering the `'` ends up escaped

Reply:
Not sure what escaping you have in mind, but experiment results suggest that it works. Well, either it works or the whole "unreleased" comment thing is not necessary.